server/user.js

```diff
@@ -0,0 +1,191 @@
+ *   confirmed: (boolean|undefined)
+ * }}
+ */
+user.userInfo;
+
+/**
+ * @typedef {{
+ *   error: (Object<{
+ *     type: string,
+ *     message: string
+ *   }>|undefined)
+ * }}
+ */
+user.ErrorResponse;
+
+/**
+ * Checks the user information for signing in.
+ * @param {string} userPath Directory of user indormation file.
+ * @param {!user.userInfo} userInfo User Information
+ * @return {user.ErrorResponse|boolean}
+ */
+user.signUp = function(userPath, userInfo) {
+  var checkDuplicate = {
+    duplicated: false,
+    elements: []
+  };
+  var lines = fs.readFileSync(userPath + user.userInfoFile).toString()
+    .split('\n');
+  for (var i = 0; i < lines.length; i++) {
+    var parts = lines[i].split(/[\t\s]+/);
+    if (parts.length == 0) {
+      continue;
+    }
+    if (parts[0] == userInfo.email || parts[1] == userInfo.username) {
+      checkDuplicate.duplicated = true;
+      if (parts[0] == userInfo.email) {
+        checkDuplicate.elements.push('email: ' + userInfo.email);
+      }
+      if (parts[1] == userInfo.username) {
+        checkDuplicate.elements.push('username: ' + userInfo.username);
+      }
+      break;
+    }
+  }
+  if (checkDuplicate.duplicated) {
+    var errorMessage = checkDuplicate.elements.join(' and ') + ' exist';
+    errorMessage += checkDuplicate.elements.length == 1 ? 's' : '';
+    return {
+      error: {
+        type: 'sign-up',
+        message: errorMessage
+      }
+    };
+  } else {
+    var infoLine = userInfo.email + ' ' + userInfo.username + ' ' +
+      userInfo.password + ' ' + userInfo.confirmed + '\n';
+    fs.appendFile(userPath + user.userInfoFile, infoLine, function(err) {
+      if (err) {
+        log.serverLog(err);
+        return;
+      }
+      log.serverLog('user information saved');
+    });
+    var folder = userPath + userInfo.username + '/';
+    if (!fs.existsSync(folder)) {
+      fs.mkdirSync(folder);
+    }
+    return true;
+  }
+};
+
+/**
+ * Checks the user information for signing in.
+ * @param {string} userPath Directory of user information file.
+ * @param {!user.userInfo} userInfo User Information
+ * @return {user.ErrorResponse|boolean}
+ */
+user.signIn = function(userPath, userInfo) {
+  var lines = fs.readFileSync(userPath + user.userInfoFile).toString()
+    .split('\n');
+  for (var i = 0; i < lines.length; i++) {
+    var parts = lines[i].split(/[\t\s]+/);
+    if (parts.length == 0) {
+      continue;
+    }
+    if (parts[1] == userInfo.username && parts[2] == userInfo.password) {
+      return true;
+    }
+  }
+  return {
+    error: {
+      type: 'sign-in',
+      message: 'invalid username or password'
+    }
+  };
+};
+
+/**
+ * Gets session ID from database. Regenerate if it is expired.
+ * @param {!mongodb.Db} db Session database.
+ * @param {string} username Username of the POST query.
+ * @param {!Object} res HTTP response.
+ * @param {function()} callback Callback function.
+ */
+user.findUserInfo = function(db, username, res, callback) {
```

Comment:
passing res does not follow the code style of the other parts of the server. Preferrably you shall return the userInfo to server.js and then respond there.